webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/pdf/util/PdfUtil.kt

```diff
@@ -120,30 +118,26 @@ object PdfUtil {
      * Fits the text to some rectangle adjusting the font size as needed.
      * @param font the font to use
      * @param text the text
-     * @param rect the rectangle where the text must fit
+     * @param availableArea the rectangle where the text must fit
      * @param maxFontSize the maximum font size
      * @param newlinesAllowed output text can be split into lines
      * @param leadingMultiplier leading multiplier between lines
      *
      * @return the calculated font size that makes the text fit
      */
-    fun fitText(font: Font, text: String, rect: Rectangle, maxFontSize: Float, newlinesAllowed: Boolean, leadingMultiplier: Float): Float {
-        // ideally, we could pass the object in which our text is going to be rendered
-        // as argument instead of asking leadingMultiplier, but we are currently rendering
-        // text in pdfcell, columntext and others
-        // it'd be painful to render lines in a common object to ask leadingMultiplier
+    fun fitText(font: Font, text: String, availableArea: Rectangle, maxFontSize: Float, newlinesAllowed: Boolean, leadingMultiplier: Float = 1f): Float {
         return binarySearchDec(1f, maxFontSize, FITTEXT_FONTSIZE_PRECISION) {
             // FIXME inplace modification is no good
             font.size = it
 
-            val lineChunks = text.splitToLineChunks(font, rect.width)
```

Comment:
@suushiemaniac this was the place that was leaking splitToLineChunks with padTurnsUniformly. First I fixed the strange behavior that scrambles were taking more than 1 page by adding padTurnsUniformly here. Then I realized I could use inside splitToLineChunks.